src/Vies/Contractor/Company/ValueAddedTax/ViesIdentifierFactory.php

```diff
@@ -0,0 +1,41 @@
+<?php
+declare(strict_types=1);
+
+namespace Landingi\BookkeepingBundle\Vies\Contractor\Company\ValueAddedTax;
+
+use DragonBe\Vies\Vies;
+use Exception;
+use Landingi\BookkeepingBundle\Bookkeeping\Contractor\Address\Country;
+use Landingi\BookkeepingBundle\Bookkeeping\Contractor\Company\ValueAddedTax\IdentifierFactory;
+use Landingi\BookkeepingBundle\Bookkeeping\Contractor\Company\ValueAddedTax\SimpleIdentifier;
+use Landingi\BookkeepingBundle\Bookkeeping\Contractor\Company\ValueAddedTax\ValidatedIdentifier;
+use Landingi\BookkeepingBundle\Bookkeeping\Contractor\Company\ValueAddedTaxIdentifier;
+use Landingi\BookkeepingBundle\Vies\ViesException;
+
+final class ViesIdentifierFactory implements IdentifierFactory
+{
+    private Vies $vies;
+
+    public function __construct(Vies $vies)
+    {
+        $this->vies = $vies;
+    }
+
+    /**
+     * @throws \Landingi\BookkeepingBundle\Vies\ViesException
+     */
+    public function create(string $identifier, string $country): ValueAddedTaxIdentifier
+    {
+        try {
+            $validation = $this->vies->validateVat($country, $identifier);
+        } catch (Exception $e) {
+            throw new ViesException('VIES external service exception: ' . $e->getMessage());
+        }
+
+        if ($validation->isValid()) {
```

Comment:
a jak nie będzie valid to ma się po prostu bez kraju zrobić?

Reply:
Tak. Bo logika jest taka jak opisana w readme (przez księgowego). Jak kraj ma aktywny VIES, to jego nip to jest ISO KRAJU + NIP, a jak nie ma to tylko NIP. i ja bym taka wartość od razu zapisywał w bazie danych. wtedy nie ma problemu z zastanawianiem się jaki identyfikator podatkowy stworzyć na fakturze

Reply:
Bardziej chodziło mi o to czy chcemy zapisywać dane, które są nieprawidłowe. Co prawda zgaduję w tym momencie, ale nie ma takiej sytuacji, żeby isValid() zwróciło Ci false dla kraju z aktywnym VIES, ale po prostu niepoprawnym identyfikatorem podatkowym? Nip też możesz podać nieprawidłowy i on przejdzie bo nie liczymy sumy kontrolnej.